src/dist/replication/lib/log_block.h

```diff
@@ -76,5 +76,62 @@ class log_block
     void init();
 };
 
+// Append writes into a buffer which consists of one or more fixed-size log blocks.
+// Not thread-safe. Requires lock protection.
+class log_appender
+{
+public:
+    explicit log_appender(int64_t start_offset)
+    {
+        _blk = std::make_shared<log_block>(start_offset);
+    }
+
+    void append_mutation(const mutation_ptr &mu, const aio_task_ptr &cb);
+
+    // Consumes one log block from buffer.
+    // This function may not take all the pending writes when there are multiple blocks.
+    std::shared_ptr<log_block> take_log_block();
+
+    size_t size() const { return _full_blocks_size + (_blk ? _blk->size() : 0); }
+    size_t blob_count() const { return _full_blocks_blob_cnt + (_blk ? _blk->data().size() : 0); }
+    bool empty() const { return !_blk; }
+
+    std::vector<mutation_ptr> mutations() const;
+
+    // Returns the heading block's start_offset.
+    int64_t start_offset() const
+    {
+        if (!_full_blocks.empty()) {
+            return (*_full_blocks.begin())->start_offset();
+        }
+        return _blk->start_offset();
+    }
+
+    // Returns the heading block's size.
+    size_t first_block_size() const
+    {
+        if (!_full_blocks.empty()) {
+            return (*_full_blocks.begin())->size();
+        }
+        return _blk->size();
+    }
+
+protected:
+    static constexpr size_t MAX_BLOCK_BYTES = 1 * 1024 * 1024; // 1MB
```

Comment:
I consider this pull request aims to restrict the size of `log_block` to avoid flushing a very large block, the default size is 1MB, could you please explain the reason why you set this value? 
Beside, will this pull request lead to any side effect? This pr will effect write operations, I just be careful about it.